### PR TITLE
Tear Gas

### PR DESCRIPTION
**Pull request by FreakingFission**
Adds Tear Gas grenades and particles, along with modifying the framework to function with tear gas

Haven't added npc blindness or a spawn icon

Would need to add npcs to the think function and avoid conflicts with flashbang blindness to allow npc blindness

### CODE REVIEW

File: lua/entities/ent_jack_gmod_ezcsparticle.lua [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+		local Time,SelfPos=CurTime(),self:GetPos()
+		if(self.DieTime<Time)then self:Remove() return end
+		local Force=VectorRand()*10
+		for key,obj in pairs(ents.FindInSphere(SelfPos,300))do
+			if(not(obj==self)and(self:CanSee(obj)))then
+				local distanceBetween = SelfPos:DistToSqr (obj:GetPos())
+				if(not obj.EZgasParticle)then
+					if((self:ShouldDamage(obj)) and self.NextDmg<Time)then
+						
+						FaceProtected = false
```
Comment (Jackarunda) [inline lua/entities/ent_jack_gmod_ezcsparticle.lua:69]: make this local
and the next one too, DEFINITELY local

Comment (FreakingFission) [reply lua/entities/ent_jack_gmod_ezcsparticle.lua:69]: make faceprotected local, but what else?

File: lua/jmod/cl_init.lua [hunk 2]
```diff
@@ -502,6 +502,22 @@ net.Receive("JMod_NuclearBlast",function()
 	end
 end)
 
+activeBlindness = 0
+targetBlindness = 0
+oldBlindness = 0
+differenceBlindness = 0
```
Comment (Jackarunda) [inline lua/jmod/cl_init.lua:508]: these can't be globals, i don't want them to be
make them values on the player, or put them all in a table or something on the player

File: lua/jmod/cl_init.lua [hunk 3]
```diff
@@ -502,6 +502,22 @@ net.Receive("JMod_NuclearBlast",function()
...
+
+net.Receive("JMod_GasBlind",function()
+	local blind = net.ReadFloat()
+	oldBlindness = targetBlindness
+	EZblindness = blind
```
Comment (Jackarunda) [inline lua/jmod/cl_init.lua:513]: another global? make sure this is scoped properly

File: lua/jmod/cl_init.lua (hunk 2 above)
Comment (FreakingFission) [reply lua/jmod/cl_init.lua:508]: Ok, ironically I did that at first

File: lua/jmod/sv_init.lua [hunk 4]
```diff
@@ -89,7 +91,23 @@ hook.Add("Think","JMOD_SERVER_THINK",function()
...
+					end
+					if (math.random(1,100)<=playa.EZblindness/1.25) then
+						JMod_TryCough(playa)
+					end
+				end
+			end
 		end
+		net.Start("JMod_GasBlind")
+		if(playa.EZblindness) then net.WriteFloat(playa.EZblindness or 0) --[[ else net.WriteFloat(0) ]] end
+		net.Send(playa)
```
Comment (Jackarunda) [inline lua/jmod/sv_init.lua:109]: gate this entirely on ezblindness, no need to waste bandwidth

Comment (FreakingFission) [reply lua/jmod/sv_init.lua:109]: What does gate mean?

File: lua/entities/ent_jack_gmod_ezcsparticle.lua (hunk 1 above)
Comment (Jackarunda) [reply lua/entities/ent_jack_gmod_ezcsparticle.lua:69]: yes, make them both local... remember that a bunch of these particles are operating together, so we don't want them interfering with eachother

File: lua/jmod/cl_hud.lua [hunk 5]
```diff
@@ -193,4 +193,42 @@ hook.Add("RenderScreenspaceEffects","JMOD_SCREENSPACE",function()
 			})
 		end
 	end
+	if(activeBlindness)then
+		if(activeBlindness > 0) then
+			DrawColorModify({
+				["$pp_colour_addr"]=0,
+				["$pp_colour_addg"]=0,
+				["$pp_colour_addb"]=0,
+				["$pp_colour_brightness"]=-(activeBlindness/111.11),
```
Comment (Jackarunda) [inline lua/jmod/cl_hud.lua:202]: clamp this to its max values that you decide

File: lua/jmod/cl_hud.lua [hunk 6]
```diff
@@ -193,4 +193,42 @@ hook.Add("RenderScreenspaceEffects","JMOD_SCREENSPACE",function()
...
+			render.UpdateScreenEffectTexture()
+			blurMaterial:SetTexture("$BASETEXTURE", render.GetScreenEffectTexture())
+			blurMaterial:SetTexture("$DEPTHTEXTURE", render.GetResolvedFullFrameDepth())
+			
+			blurMaterial:SetFloat("$size", (activeBlindness/10))
+			blurMaterial:SetFloat("$focus", 1)
+			blurMaterial:SetFloat("$focusradius", 2)
+			
+			render.SetMaterial(blurMaterial)
+			render.DrawScreenQuad()
```
Comment (Jackarunda) [inline lua/jmod/cl_hud.lua:219]: i'd really like to see this as just drawToyTown or drawMotionBlur, not this quad thing, unless you can describe to me why you chose this method

**Review by Jackarunda — APPROVED** [review]

**Pull request merged**